src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {useEffect, useRef} from 'react';
+import Nav from 'rsuite/lib/Nav/index.js';
+import styles from '../styles/sidebar.module.css';
+
+const ITEM_HEIGHT = 42;
+const SIDEBAR_HEIGHT = 300;
+
+let timer;
+
+export default function Sidebar({section, onChange, providers}) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (section.eventKey == null) return;
+
+    clearTimeout(timer);
+
+    timer = setTimeout(() => {
+      const top = containerRef.current.scrollTop;
+      const index = providers.findIndex((provider) => provider.id === section.eventKey);
+      const depth = index * ITEM_HEIGHT;
+
+      if (depth < top) {
+        containerRef.current.scrollTo({
+          top: depth,
+          left: 0,
+          behavior: 'smooth',
+        });
+
+        return;
+      }
+
+      if (depth + ITEM_HEIGHT > top + SIDEBAR_HEIGHT) {
+        containerRef.current.scrollTo({
+          top: depth - SIDEBAR_HEIGHT + 36,
+          left: 0,
+          behavior: 'smooth',
+        });
+      }
+    }, 100);
```

Comment:
i think you're intending to use this as a debounce, but this forces a required delay on changes. I think we'd want to use a throttle library for this, which allows us to call this logic immediately, and then again after function debounce.